packages/astro/src/core/render/result.ts

```diff
@@ -94,21 +95,27 @@ class Slots {
 		if (!this.has(name)) return undefined;
 		if (!cacheable) {
 			const component = await this.#slots[name]();
-			const expression = getFunctionExpression(component);
-
 			if (!Array.isArray(args)) {
 				warn(
 					this.#loggingOpts,
 					'Astro.slots.render',
 					`Expected second parameter to be an array, received a ${typeof args}. If you're trying to pass an array as a single argument and getting unexpected results, make sure you're passing your array as a item of an array. Ex: Astro.slots.render('default', [["Hello", "World"]])`
 				);
 			} else {
+				// Astro
+				const expression = getFunctionExpression(component);
 				if (expression) {
 					const slot = expression(...args);
 					return await renderSlot(this.#result, slot).then((res) =>
 						res != null ? String(res) : res
 					);
 				}
+				// JSX
+				if (typeof component === 'function') {
+					return await renderJSX(this.#result, component(...args)).then((res) =>
+						res != null ? String(res) : res
+					);
+				}
```

Comment:
This mixes JSX rendering into `Astro.slots.render`. I'm not sure if this is fine architecture-wise.

Reply:
Yeah, that this is a bit odd but I can't think of a better way. I think it's ok for now.